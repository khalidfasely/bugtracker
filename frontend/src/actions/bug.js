import bugApi from "../api/bug";

export const setBug = (bug) => ({
    type: "SET_BUG",
    bug
});

export const startSetBug = (pid, bid) => {
    return (dispatch) => {
        return bugApi(pid, bid).then((result) => {
            dispatch(setBug(result.bug));
            return result;
        });
    };
};