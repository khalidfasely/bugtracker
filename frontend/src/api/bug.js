export default async (pid, bid) => {
    const res = await fetch(`/api/project/${pid}/bug/${bid}`);

    const result = await res.json();

    return result;
};