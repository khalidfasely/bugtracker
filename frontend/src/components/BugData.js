const BugData = ({ bug }) => (
    <div>
        <p>{bug?.id}</p>
        <p>{bug?.user?.username}</p>
        <p>{bug?.title}</p>
        <p>{bug?.description}</p>
        <p>{bug?.time}</p>
        <p>{bug?.classification}</p>
        <p>{`${bug?.active}`}</p>
        <div>
            {bug?.admins?.map(admin => <span key={admin} >{admin} </span>)}
        </div>
        <br />
        <div>
            {bug?.users_with?.map(user => <span key={user} >{user} </span>)}
        </div>
    </div>
);

export default BugData;