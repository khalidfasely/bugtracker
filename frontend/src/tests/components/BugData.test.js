import React from "react";
import { render } from "@testing-library/react";

import bug from "../fixtures/bug";
import BugData from "../../components/BugData";

test('Should render BugData component correctly without bug data', () => {
    const { asFragment } = render(<BugData />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should render BugData component correctly with bug data', () => {
    const props = {
        bug
    }
    const { asFragment } = render(<BugData {...props} />);
    expect(asFragment()).toMatchSnapshot();
});