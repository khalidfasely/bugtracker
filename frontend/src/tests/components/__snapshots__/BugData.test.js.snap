// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BugData component correctly with bug data 1`] = `
<DocumentFragment>
  <div>
    <p>
      1
    </p>
    <p>
      admin
    </p>
    <p>
      Bug title
    </p>
    <p>
      Bug description
    </p>
    <p>
      now
    </p>
    <p>
      medium
    </p>
    <p>
      true
    </p>
    <div>
      <span>
        admin 
      </span>
      <span>
        1 
      </span>
    </div>
    <br />
    <div>
      <span>
        admin 
      </span>
      <span>
        1 
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should render BugData component correctly without bug data 1`] = `
<DocumentFragment>
  <div>
    <p />
    <p />
    <p />
    <p />
    <p />
    <p />
    <p>
      undefined
    </p>
    <div />
    <br />
    <div />
  </div>
</DocumentFragment>
`;
